index.js

```diff
@@ -31,38 +95,91 @@ module.exports = async (options = {}) => {
 		}
 	}
 
-	const defaultHeaders = [
-		'imageName',
-		'pid',
-		'sessionName',
-		'sessionNumber',
-		'memUsage'
-	];
-
-	const verboseHeaders = defaultHeaders.concat([
-		'status',
-		'username',
-		'cpuTime',
-		'windowTitle'
-	]);
+	let currentHeader;
+	if (options.apps) {
+		currentHeader = 'apps';
+	} else if (options.modules !== undefined) {
+		currentHeader = 'modules';
+	} else if (options.services) {
+		currentHeader = 'services';
+	} else {
+		currentHeader = 'default';
+	}
 
-	const headers = options.verbose ? verboseHeaders : defaultHeaders;
+	if (options.verbose) {
+		currentHeader += 'Verbose';
+	}
 
-	const {stdout} = await execFile('tasklist', args);
+	const columns = csvHeaders[currentHeader];
+	const currentTransform = transforms[currentHeader];
+	return {args, columns, currentTransform};
+}
 
-	// Not start with `"` means no matching tasks. See #11.
-	const data = stdout.startsWith('"') ? await neatCsv(stdout, {headers}) : [];
+/**
+ * Execute tasklist and return results through a stream
+ * @param {Options} options The options of the command
+ * @returns {Stream} Stream, returning parsed results
+ */
+function stream(options = {}) {
+	const {args, columns, currentTransform} = main(options);
+	const parser = csv.parse({columns});
+	const pt = passThrough();
+	const processOutput = childProcess.spawn('tasklist.exe', args).stdout;
+	let outputChecked = false;
 
-	return data.map(task => {
-		// Normalize task props
-		task.pid = Number(task.pid);
-		task.sessionNumber = Number(task.sessionNumber);
-		task.memUsage = Number(task.memUsage.replace(/[^\d]/g, '')) * 1024;
+	processOutput.on('data', data => {
```

Comment:
I'm not sure that this is a good way to implement the streaming. You're manually connecting your streams rather that `.pipe`'ing the data through them. I think that this means that, for example, once the returned stream is aborted, the other streams will continue to handle incoming tasklist data, rather than being aborted, too.

Haven't worked with Node streams in a few years, though, so I might be wrong.

@sindresorhus Any input on this?